bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -328,4 +352,41 @@ private void refreshChannelSubscription(ChannelUID channelUID) {
     public Collection<Class<? extends ThingHandlerService>> getServices() {
         return Collections.singleton(LGWebOSActions.class);
     }
+
+    @Override
+    public void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
```

Comment:
Take care, you have to filter to not run your code for any thing discovered.

Reply:
you are right about filtering to avoid this happening too often

if we put this into initialize it may execute even before the TV appears on the network. 
there may be situations when arp is not yet filled.

hm, on the other hand, it is optional to detect the mac, and it typically only needs to run once. 

have prepared the code change, but need to run some tests

Reply:
I was thinking during the init process, not necessarly in initialize method. I let you check but maybe you can do that after the web socket connection is established ?

Reply:
Ok, pushed a new version. This has DiscoveryListener removed as well.
MAC address detection in thingDiscovered has moved into initialize and after websocket connect phase.
thingRemoved was called at the same time as websocket disconnect, so no point in keeping it.
Pls have a look...